varnish-macros/src/stats.rs

```diff
@@ -0,0 +1,162 @@
+use crate::parser_utils::parse_doc_str;
+use serde::Serialize;
+use syn::{punctuated::Punctuated, token::Comma, Data, Field, Fields, Type};
+
+type FieldList = Punctuated<Field, Comma>;
+
+#[derive(Serialize, Clone)]
+struct VscMetricDef {
+    pub name: String,
+    #[serde(rename = "type")]
+    pub counter_type: String, // "counter", "gauge"
+    pub ctype: String,    // "uint64_t" is only option right now
+    pub level: String,    // "info", "debug", etc
+    pub oneliner: String, // "Counts the number of X", etc
+    pub format: String,   // "integer", "bytes", "duration", "bitmap", etc
+    pub docs: String,
+    pub index: Option<usize>,
+}
+
+#[derive(Serialize)]
+struct VscMetadata {
+    version: String,
+    name: String,
+    oneliner: String,
+    order: u32,
+    docs: String,
+    elements: usize,
+    elem: std::collections::HashMap<String, VscMetricDef>,
+}
+
+pub fn get_struct_fields(data: &Data) -> &FieldList {
+    match data {
+        Data::Struct(data) => match &data.fields {
+            Fields::Named(fields) => &fields.named,
+            _ => panic!("Only named fields are supported"),
+        },
+        _ => panic!("Only structs are supported"),
+    }
+}
+
+pub fn validate_fields(fields: &FieldList) {
+    for field in fields {
+        match &field.ty {
+            Type::Path(path) => {
+                let is_atomic_u64 = path
+                    .path
+                    .segments
+                    .last()
+                    .is_some_and(|seg| seg.ident == "AtomicU64");
+
+                if !is_atomic_u64 {
+                    let field_name = field.ident.as_ref().unwrap();
+                    panic!("Field {field_name} must be of type AtomicU64");
+                }
+            }
+            _ => panic!("Field types must be AtomicU64"),
+        }
+    }
+}
+
+fn generate_metrics(fields: &FieldList) -> Vec<VscMetricDef> {
+    fields
+        .iter()
+        .enumerate()
+        .map(|(i, field)| {
+            let name = field.ident.as_ref().unwrap().to_string();
+
+            let counter_type = if field
```

Comment:
Let's create a utility function like `has_attr` and put it next to `remove_attr` in the `parser_utils.rs`.  TIL you can remove certain attributes automatically by adding them to the `proc_macro_derive` as a second parameter - I had to remove them because i didn't know it existed